google/resource_storage_bucket_iam_test.go

```diff
@@ -86,6 +115,71 @@ func testAccCheckGoogleStorageBucketIam(bucket, role string, members []string) r
 	}
 }
 
+func testAccStorageBucketIamPolicy_update(bucket, account string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_service_account" "test-account-1" {
+  account_id   = "%s-1"
+  display_name = "Iam Testing Account"
+}
+
+resource "google_service_account" "test-account-2" {
+  account_id   = "%s-2"
+  display_name = "Iam Testing Account"
+}
+
+
+data "google_iam_policy" "fooPolicy" {
+  binding {
+    role = "roles/storage.objectViewer"
+
+    members = [
+      "serviceAccount:${google_service_account.test-account-1.email}",
+      "serviceAccount:${google_service_account.test-account-2.email}"
+    ]
+  }
+}
+
+resource "google_storage_bucket_iam_policy" "foo" {
+  bucket      = "${google_storage_bucket.%s}"
+  policy_data = "${data.google_iam_policy.fooPolicy.policy_data}"
+}
+
+`, bucket, account, account, bucket)
+}
+
+func testAccStorageBucketIamPolicy_basic(bucket, account string) string {
+	return fmt.Sprintf(`
+resource "google_storage_bucket" "bucket" {
+  name = "%s"
+}
+
+resource "google_service_account" "test-account-1" {
+  account_id   = "%s-1"
+  display_name = "Iam Testing Account"
+}
+
+data "google_iam_policy" "fooPolicy" {
+  binding {
+    role = "roles/storage.objectViewer"
+
+    members = [
+      "serviceAccount:${google_service_account.test-account-1.email}",
+    ]
+  }
+}
+
+resource "google_storage_bucket_iam_policy" "fooPolicy" {
+  bucket      = "${google_storage_bucket.%s}"
```

Comment:
This is the problem - you'll need to append `.name`.  You can't use the entire resource here, you need to access the right attribute on the resource, which is `.name`.